src/transaction/TransactionController.test.ts

```diff
@@ -606,28 +606,27 @@ describe('TransactionController', () => {
   });
 
   it('should cancel a transaction', async () => {
-    await new Promise(async (resolve) => {
-      const controller = new TransactionController({ provider: PROVIDER });
-      controller.context = {
-        NetworkController: MOCK_NETWORK,
-      } as any;
-      controller.onComposed();
-      const from = '0xc38bf1ad06ef69f0c04e29dbeb4152b4175f0a8d';
-      const { result } = await controller.addTransaction({
-        from,
-        to: from,
-      });
-      controller.cancelTransaction('foo');
+    const controller = new TransactionController({ provider: PROVIDER });
+    controller.context = {
+      NetworkController: MOCK_NETWORK,
+    } as any;
+    controller.onComposed();
+    const from = '0xc38bf1ad06ef69f0c04e29dbeb4152b4175f0a8d';
+    const { result } = await controller.addTransaction({
+      from,
+      to: from,
+    });
+    controller.cancelTransaction('foo');
+    const transactionListener = new Promise(async (resolve) => {
       controller.hub.once(`${controller.state.transactions[0].id}:finished`, () => {
         expect(controller.state.transactions[0].transaction.from).toBe(from);
         expect(controller.state.transactions[0].status).toBe(TransactionStatus.rejected);
-      });
-      controller.cancelTransaction(controller.state.transactions[0].id);
-      result.catch((error) => {
-        expect(error.message).toContain('User rejected the transaction');
         resolve('');
```

Comment:
Nit: we don't need to pass anything to `resolve` here.

```suggestion
        resolve();
```

Reply:
Bah. I committed this, but now the test fails because the type needs to be adjusted. I'm going to undo this for now - we can fix it some other time.